coffea/processor/executor.py

```diff
@@ -817,28 +1029,143 @@ def __call__(
+                            accumulate(
+                                progress.track(
+                                    map(_decompress, (c for c in batch)),
+                                    task_id=p_id_m,
+                                    total=progress._tasks[p_id_m].total + len(batch),
+                                ),
+                                _decompress(merged),
+                            ),
+                            self.compression,
+                        )
+                # Add checkpointing
+
+                if self.merging:
+                    merged = FH.completed.pop().result()
+                if len(FH.completed) > 0:
+                    raise RuntimeError("Not all futures are added.")
+                return accumulate([_decompress(merged), accumulator]), 0
+
+        except Exception as e:
+            if self.recoverable:
+                print(f"Exception '{type(e)}' occured, recovering progress...")
+                # for job in FH.futures:  # NotImplemented in parsl
+                #     job.cancel()
+
+                with rich_bar() as progress:
+                    if self.merging:
+                        prog_id_wait = progress.add_task(
+                            "Waiting for merge jobs",
+                            total=len(FH.merges),
+                            unit=self.unit,
+                        )
+                        while len(FH.merges) > 0:
+                            FH.update()
+                            progress.update(
+                                prog_id_wait,
+                                completed=(
+                                    progress._tasks[prog_id_wait].total - len(FH.merges)
+                                ),
+                                refresh=True,
+                            )
+
+                    def is_good(future):
+                        return (
+                            future.done()
+                            and not future.cancelled()
+                            and future.exception() is None
+                        )
+
+                    FH.update()
+                    recovered = [
+                        future.result() for future in FH.completed if is_good(future)
+                    ]
+                    p_id_m = progress.add_task("Merging finished jobs", unit="merges")
+                    merged = _compress(
+                        accumulate(
+                            progress.track(
+                                map(_decompress, (c for c in recovered)),
+                                task_id=p_id_m,
+                                total=len(recovered),
+                            )
+                        ),
+                        self.compression,
+                    )
 
-        return accumulator
+                    return accumulate([_decompress(merged), accumulator]), e
+            else:
+                raise type(e)(str(e)).with_traceback(sys.exc_info()[2]) from None
```

Comment:
Based on #650 this should just be `raise Exception(...) from e`

Reply:
👍